CONTRIBUTING.md

```diff
@@ -34,6 +34,37 @@ Additionally, it may be necessary to determine build failures, these jobs are lo
 If a package continuously fails to build, the ROS Boss for that distribution may choose to revert the pull request that introduced it.
 Or they may take [other actions](https://github.com/ros-infrastructure/ros_buildfarm/blob/master/doc/ongoing_operations.rst) to avoid repeated failures.
 
+### Releasing third party packages into a ROS distribution
+
+In general you should not override or replace system dependencies from the system distribution with a package in the ROS distribution.
+This is part of the software engineering process of maintaining the distribution that all developers can rely on specific versions of software to be available and remain consistent withing the distribution.
+If you need newer versions of underlying libraries the you should either target newer versions with the minimum version available or if that’s not available help and encourage the upstream process to move forward to at least your minimum version for the next release of the distribution.
+
+Obviously this is a slow process and doesn’t fix things immediately for a short term there are workarounds, however if you want to release packages which build on the workarounds you must make sure not to break other users who are expecting the system version.
```

Comment:
```suggestion
Obviously this is a slow process and doesn’t fix things immediately for the short term.  If you install the updated package locally and introduce workarounds into your package, you must make sure not to break other users who are expecting the system version when you release the package.
```

Reply:
Hmm, clearly not a properly formed sentence. Here's a slightly more complete suggestion.
```suggestion
Obviously this is a slow process and doesn’t fix things immediately for a short term.
There are workarounds such as embedding copies of the libraries into your package or making another package with the newer library version.
This usually can be made to work well in your local development environment.
However if you want to release this style of workarounds you must make sure not to break other users who are expecting the system version either in downstream packages or elsewhere on the system.
```